sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/metric/DistributionStatisticConfig.java

```diff
@@ -0,0 +1,206 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.sermant.core.service.metric;
+
+import java.time.Duration;
+
+/**
+ * DistributionStatisticConfig
+ *
+ * @author zwmagic
+ * @since 2024-08-17
+ */
+public class DistributionStatisticConfig {
+    private Boolean percentileHistogram;
+
+    private double[] percentiles;
+
+    private Integer percentilePrecision;
+
+    private Double minimumExpectedValue;
+
+    private Double maximumExpectedValue;
+
+    private Duration expiry;
+
+    private Integer bufferLength;
+
+    /**
+     * Gets the percentile histogram flag.
+     *
+     * @return the percentile histogram flag
+     */
+    public Boolean getPercentileHistogram() {
+        return percentileHistogram;
+    }
+
+    /**
+     * Gets the array of percentiles.
+     *
+     * @return the array of percentiles
+     */
+    public double[] getPercentiles() {
+        return percentiles;
+    }
+
+    /**
+     * Gets the precision for percentiles.
+     *
+     * @return the precision for percentiles
+     */
+    public Integer getPercentilePrecision() {
+        return percentilePrecision;
+    }
+
+    /**
+     * Gets the minimum expected value.
+     *
+     * @return the minimum expected value
+     */
+    public Double getMinimumExpectedValue() {
+        return minimumExpectedValue;
+    }
+
+    /**
+     * Gets the maximum expected value.
+     *
+     * @return the maximum expected value
+     */
+    public Double getMaximumExpectedValue() {
+        return maximumExpectedValue;
+    }
+
+    /**
+     * Gets the expiry duration.
+     *
+     * @return the expiry duration
+     */
+    public Duration getExpiry() {
+        return expiry;
+    }
+
+    /**
+     * Gets the length of the buffer.
+     *
+     * @return the length of the buffer
+     */
+    public Integer getBufferLength() {
+        return bufferLength;
+    }
+
+    /**
+     * Produces an additional time series for each requested percentile. This percentile is computed locally, and so
+     * can't be aggregated with percentiles computed across other dimensions (e.g. in a different instance). Use
+     * {@link #publishPercentileHistogram()} to publish a histogram that can be used to generate aggregable percentile
+     * approximations.
+     *
+     * @param percentilesValue Percentiles to compute and publish. The 95th percentile should be expressed as
+     * {@code 0.95}.
+     * @return This builder.
+     */
+    public DistributionStatisticConfig publishPercentiles(double[] percentilesValue) {
+        this.percentiles = percentilesValue;
+        return this;
+    }
```

Comment:
Note that if you quote refer or copy codes from other open source project，you have to declar it in Copyrighnt

Reply:
ok